plugins/net_plugin/net_plugin.cpp

```diff
@@ -2529,20 +2543,24 @@ namespace eosio {
                } else {
                   // use chain head instead of fork head so we do not get too far ahead of applied blocks
                   uint32_t head = my_impl->get_chain_head_num();
```

Comment:
Isn't using chain_head here a problem, as fork_db head can be higher?

Reply:
`fork_db` `head` will always be >= `chain_head`. Therefore, `head + fetch_span`  < `forkdb head + fetch_span`.

Reply:
exactly. But we are concerned about `forkdb head + fetch_span` becoming greater than `forkdb.size()`.

Reply:
Example:
```c++
chain_head = 1000
fetch_span = 2000
fork_db.size() = 3000
fork_db.head = 4000

sync received block (blk_num) = 3001

// blk_num < chain_head + fetch_span 
if (3001 < 1000 + 2000) // false
   request more

// blk_num < fork_db.head + fetch_span
if (3001 < 4000 + 2000) // true
   request more
```

Therefore, we are more constrained to "request more" by using chain head.

Reply:
Yes, thanks, I totally misread that!